### PR TITLE
Gh424rework

### PR DESCRIPTION
**Pull request by MarkMLunarG**
Please see #424 for omissions,
discrepancies, duplications,
disparities, and deralictions.

### CODE REVIEW

File: layers/vk_validation_layer_details.md [hunk 1]
```diff
@@ -90,7 +90,7 @@ The Draw State portion of the core validation layer tracks state leading into Dr
...
-| Verify Get Queries| Validate that that queries are properly setup, initialized and synchronized | INVALID_QUERY | vkGetFenceStatus vkQueueWaitIdle vkWaitForFences vkDeviceWaitIdle vkCmdBeginQuery vkCmdEndQuery | TODO | Need to check existing case against object_tracker and remove any redundant checks. Then write tests for remaining case. Currently there are 8 cases for this check with 1 each in cleanInFlightCmdBuffer(), EndCommandBuffer(), CmdEndQuery(), validateQuery(), and 4 cases in GetQueryPoolResults() |
+| Verify Get Queries| Validate that that queries are properly setup, initialized and synchronized | INVALID_QUERY | vkGetFenceStatus vkQueueWaitIdle vkWaitForFences vkDeviceWaitIdle vkCmdBeginQuery vkCmdEndQuery | InvalidQueueIndexInvalidQuery | May need to check existing case against object_tracker and remove any redundant checks. Then write tests for remaining case. Currently there are 8 cases for this check with 1 each in cleanInFlightCmdBuffer(), EndCommandBuffer(), CmdEndQuery(), validateQuery(), and 4 cases in GetQueryPoolResults() |
```
Comment (tobine) [inline layers/vk_validation_layer_details.md]: duplicate "that that" - not introduced by you, but would be nice to clean up while you're touching this line

**Comment by MarkMLunarG** [comment]
The latest commit is to get rid of that that scat

File: tests/layer_validation_tests.cpp [hunk 2]
```diff
@@ -12182,6 +12182,147 @@ TEST_F(VkLayerTest, VtxBufferBadIndex) {
...
+            invalid_feature_found = true;
+            break;
+        }
+    }
+
+    if (invalid_feature_found) {
+        m_errorMonitor->SetDesiredFailureMsg(VK_DEBUG_REPORT_ERROR_BIT_EXT,
+                                             unavailable_feature_message);
+        device_create_info.pEnabledFeatures = &features;
+        vkCreateDevice(gpu(), &device_create_info, nullptr, &testDevice);
```
Comment (tobine) [inline tests/layer_validation_tests.cpp]: It appears that at this point in the test the invalid queueFamilyIndex is still set. Would like to see it updated to a supported value prior to the final test case for clarity.

File: tests/layer_validation_tests.cpp [hunk 3]
```diff
@@ -12182,6 +12182,147 @@ TEST_F(VkLayerTest, VtxBufferBadIndex) {
...
+    img_barrier.srcAccessMask = VK_ACCESS_HOST_WRITE_BIT;
+    img_barrier.dstAccessMask = VK_ACCESS_SHADER_READ_BIT;
+    img_barrier.oldLayout = VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL;
+    img_barrier.newLayout = VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL;
+    img_barrier.image = image.handle();
+    img_barrier.srcQueueFamilyIndex = VK_QUEUE_FAMILY_IGNORED;
+
+    // QueueFamilyIndex must be VK_QUEUE_FAMILY_IGNORED, this verifies
+    // that layer validation catches the case when it is not.
+    //img_barrier.dstQueueFamilyIndex = VK_QUEUE_FAMILY_IGNORED;
```
Comment (tobine) [inline tests/layer_validation_tests.cpp]: Minor thing but having this line commented out is not completely clear. Would rather see dstQueueFamilyIndex assigned explicitly to "0". I know that's what it is, but setting explicitly would make test more clear and comment could note that "0" is being forced to make src/dst mismatch to cause error.

**Comment by tobine** [comment]
Just a couple minor updates for clarity. Otherwise passes on our farm and
LGTM :+1:

**Comment by MarkMLunarG** [comment]
SHA-1: ec9e24b63c7ac85bf665992ac1405182288b6fcd

* docs: GH424 Update validation test documentation

SHA-1: e63035786ab9f202cc16425ca86d57434b9cd456

* tests: GH424 Addition of Device Limits Validation tests

SHA-1: 1d659b4113b77a95325df10d602a03f1e7abf8b7

* layers: GH424 add check of skip_call results to CreateDevice

**Pull request merged**